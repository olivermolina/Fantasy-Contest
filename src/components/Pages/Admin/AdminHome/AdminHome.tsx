import React from 'react';
import { NextRouter } from 'next/router';
import Grid from '@mui/material/Unstable_Grid2';
import List from '@mui/material/List';
import ListSubheader from '@mui/material/ListSubheader';
import ListItemButton from '@mui/material/ListItemButton';
import ListItemText from '@mui/material/ListItemText';
import Divider from '@mui/material/Divider';
import type { AdminMenuType } from '~/server/routers/admin/getMenus';

interface Props {
  router: NextRouter;
  menus: AdminMenuType[];
}

export default function AdminHome(props: Props) {
  const { router, menus } = props;
  return (
    <Grid container rowSpacing={1} columnSpacing={{ xs: 1, sm: 2, md: 3 }}>
      {menus.map((menu) => (
        <Grid xs={12} md={6} lg={4} key={menu.category}>
          <List
            sx={{ width: '100%', bgcolor: 'background.paper' }}
            component="nav"
            aria-labelledby={menu.category}
            subheader={
              <ListSubheader
                id={menu.category}
                sx={{
                  fontSize: 25,
                  fontWeight: 'bold',
                }}
                className={'border-b-8 border-blue-200 '}
              >
                {menu.category}
              </ListSubheader>
            }
          >
            {menu.subMenus.map((submenu) => (
              <React.Fragment key={submenu.urlPath}>
                <ListItemButton onClick={() => router.push(submenu.urlPath)}>
                  <ListItemText primary={submenu.label} />
                </ListItemButton>
                <Divider />
              </React.Fragment>
            ))}
          </List>
        </Grid>
      ))}
    </Grid>
  );
}
