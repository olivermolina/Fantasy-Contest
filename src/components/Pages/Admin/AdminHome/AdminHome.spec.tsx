import React from 'react';
import { render } from '@testing-library/react';
import AdminHome from './AdminHome';
import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));
describe('AdminHome', () => {
  it('should render', () => {
    const router = useRouter();
    const { container } = render(<AdminHome router={router} menus={[]} />);
    expect(container).toMatchSnapshot();
  });
});
