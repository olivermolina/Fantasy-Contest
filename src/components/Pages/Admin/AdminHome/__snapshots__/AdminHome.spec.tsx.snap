// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminHome should render 1`] = `
<div>
  <div
    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-sftz05-MuiGrid2-root"
  />
</div>
`;
